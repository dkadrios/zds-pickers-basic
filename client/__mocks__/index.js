export inputControls from './inputControls'
export mappings from './mappings'
export midi from './midi'
export shiftGroups from './shiftGroups'
export shifter from './shifter'
export store from './store'
export user from './user'
export version from './version'
