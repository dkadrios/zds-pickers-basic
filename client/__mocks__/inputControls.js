export default [
  {
    controlId: 0,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 0,
    status: 11,
    value: 110,
    active: true,
    lit: true,
  },
  {
    controlId: 1,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 1,
    status: 11,
    value: 111,
    active: false,
    lit: false,
  },
  {
    controlId: 2,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 0,
    status: 9,
    value: 110,
    active: true,
    lit: false,
  },
  {
    controlId: 3,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 0,
    status: 9,
    value: 110,
    active: true,
    lit: true,
  },
  {
    controlId: 4,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 0,
    status: 9,
    value: 110,
    active: true,
    lit: true,
  },
  {
    controlId: 5,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 0,
    status: 9,
    value: 110,
    active: true,
    lit: false,
  },
]
