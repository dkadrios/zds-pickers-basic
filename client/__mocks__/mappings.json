{
  "dialogVisible": false,
  "userDialogVisible": false,
  "userImportDialogVisible": false,
  "channels": [
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    "BFD3",
    null,
    null,
    null,
    null,
    null,
    null
  ],
  "stockMappings": [
    "2box",
    "Addictive Drums",
    "Alesis DM Pro",
    "Alesis DM10",
    "Alesis DM5",
    "BFD2",
    "BFD3",
    "Battery 3",
    "E-MU Protean Acoustic",
    "E-MU Protean Bread n Butter",
    "E-MU Protean Studio",
    "E-MU Protean Synth",
    "EZ Drummer The Classic EZX",
    "EZDrummer",
    "EZDrummer Jazz Kit",
    "EZDrummer Latin Percussion",
    "General MIDI",
    "Infinite Instruments",
    "John Emrich - Articulations 1",
    "Kontakt 2",
    "Ocean Way Drums",
    "Pearl Redbox",
    "Roland HandSonic 15",
    "Roland TD-10",
    "Roland TD-10 Perc 1",
    "Roland TD-10 Perc 2",
    "Roland TD-10 Perc 3",
    "Roland TD-10 Perc 4",
    "Roland TD-12",
    "Roland TD-12 Perc 1 (Latin)",
    "Roland TD-12 Perc 2 (Indian)",
    "Roland TD-12 Perc 3 (African)",
    "Roland TD-12 Perc 4 (Salsa)",
    "Roland TD-12 Perc 5 (SFX)",
    "Roland TD-12 Perc 6-8 (User)",
    "Roland TD-20",
    "Roland TD-20 Perc",
    "Roland TD-3",
    "Roland TD-30",
    "Roland TD-30 Perc 1 (Latin)",
    "Roland TD-30 Perc 2 (Indian)",
    "Roland TD-30 Perc 3 (African)",
    "Roland TD-30 Perc 4 (Salsa)",
    "Roland TD-30 Perc 5 (SFX)",
    "Roland TD-30 Perc 6-8 (User)",
    "Roland TD-5",
    "Roland TD-6",
    "Roland TD-6 Perc 1",
    "Roland TD-6 Perc 10 (Special)",
    "Roland TD-6 Perc 2",
    "Roland TD-6 Perc 3 (Room)",
    "Roland TD-6 Perc 4 (Power)",
    "Roland TD-6 Perc 5 (Elec)",
    "Roland TD-6 Perc 6 (808)",
    "Roland TD-6 Perc 7 (Jazz)",
    "Roland TD-6 Perc 8 (Brush)",
    "Roland TD-6 Perc 9 (Perc)",
    "Roland TD-7",
    "Roland TD-8",
    "Roland TD-8 Perc 1",
    "Roland TD-8 Perc 10 (Special)",
    "Roland TD-8 Perc 2",
    "Roland TD-8 Perc 3 (Room)",
    "Roland TD-8 Perc 4 (Power)",
    "Roland TD-8 Perc 5 (Elec)",
    "Roland TD-8 Perc 6 (808)",
    "Roland TD-8 Perc 7 (Jazz)",
    "Roland TD-8 Perc 8 (Brush)",
    "Roland TD-8 Perc 9 (Perc)",
    "Roland TD-9",
    "Roland TD-9 Perc",
    "Simmons SD1000",
    "Simmons SD5X",
    "Simmons SD7",
    "Simmons SD7PK",
    "Simmons SD9K",
    "Sonic Reality iMap Extended",
    "Sonic Reality iMap Standard",
    "Studio ProFiles",
    "Superior Drummer",
    "Yamaha DTX 700",
    "Yamaha DTXplorer",
    "Zendrum Stompblock",
    "Zendrum TRS",
    "Zendrum TRS 2.0"
  ],
  "userMappings": ["BFD3 Custom ZenEdit", "reaper"]
}
