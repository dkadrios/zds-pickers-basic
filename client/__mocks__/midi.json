{
  "devices": [
    {
      "id": "-1325431060",
      "manufacturer": "Zendrum Studio",
      "name": "ZDS Shifter",
      "type": "output",
      "version": "",
      "state": "connected",
      "connection": "open"
    },
    {
      "id": "-312145282",
      "manufacturer": "Zendrum Studio",
      "name": "ZDS Shifter",
      "type": "input",
      "version": "",
      "state": "connected",
      "connection": "open"
    }
  ],
  "listeningDevices": ["-312145282"]
}
