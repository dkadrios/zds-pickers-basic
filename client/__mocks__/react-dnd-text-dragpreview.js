/* eslint-disable import/prefer-default-export, jsx-a11y/alt-text */
import React from 'react'

export const createDragPreview = () => <img src={null} />
