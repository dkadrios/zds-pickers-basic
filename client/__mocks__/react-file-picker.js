/* eslint-disable import/prefer-default-export */
export const FilePicker = () => <div>content</div>
