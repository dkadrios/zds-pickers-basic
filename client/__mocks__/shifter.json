{
  "accessGranted": true,
  "errorMessage": "",
  "errorVisible": false,
  "exportBuffer": [],
  "exportDialogVisible": false,
  "exportFilename": "zds-shifter-backup.txt",
  "found": true,
  "hardwareTestVisible": true,
  "importDialogVisible": false,
  "importInProcess": false,
  "invalidSettingsFile": null,
  "midiActivityLEDMode": 0,
  "midiInActivity": false,
  "midiOutActivity": false,
  "ready": true,
  "resetInProcess": false,
  "responding": true,
  "searchedForShifter": true,
  "serialMidiOutEnabled": true,
  "showResetDialog": false,
  "testInterfaceFound": false,
  "usbMidiOutEnabled": true
}
