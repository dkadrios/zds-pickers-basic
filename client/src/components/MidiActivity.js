import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import {
  ACTIVITY_LED_MODE_ALWAYS_OFF,
  ACTIVITY_LED_MODE_ALWAYS_ON,
  ACTIVITY_LED_MODE_NORMALLY_OFF,
  ACTIVITY_LED_MODE_NORMALLY_ON,
} from '../midi/sysex'
import { shifterShape } from '../core/shapes'
import Led from './Led'

export const isLit = ({ midiInActivity, midiOutActivity, midiActivityLEDMode }) => {
  switch (midiActivityLEDMode) {
    case ACTIVITY_LED_MODE_NORMALLY_ON:
      return !midiInActivity && !midiOutActivity
    case ACTIVITY_LED_MODE_NORMALLY_OFF:
      return midiInActivity || midiOutActivity
    case ACTIVITY_LED_MODE_ALWAYS_ON:
      return true
    case ACTIVITY_LED_MODE_ALWAYS_OFF:
    default:
      return false
  }
}

export const MidiActivity = ({ shifter, label }) => (
  <div style={{ margin: '8px 7px 0 0' }}>
    <Led
      isMidiActivity
      label={label}
      lit={isLit(shifter)}
    />
  </div>
)

MidiActivity.propTypes = {
  label: PropTypes.string.isRequired,
  shifter: shifterShape.isRequired,
}

export const mapStateToProps = ({ shifter }) => ({ shifter })

export default connect(mapStateToProps)(MidiActivity)
