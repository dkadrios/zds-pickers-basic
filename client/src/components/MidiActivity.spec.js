import { shifter } from '../../__mocks__'
import {
  ACTIVITY_LED_MODE_ALWAYS_OFF,
  ACTIVITY_LED_MODE_ALWAYS_ON,
  ACTIVITY_LED_MODE_NORMALLY_OFF,
  ACTIVITY_LED_MODE_NORMALLY_ON,
} from '../midi/sysex'
import { MidiActivity, isLit, mapStateToProps } from './MidiActivity'

describe('MidiActivity tests', () => {
  const props = {
    shifter,
    label: 'label',
  }

  it('renders correctly', () => {
    shallowExpect(MidiActivity)(props).toMatchSnapshot()
  })

  it('lights up correctly for ACTIVITY_LED_MODE_NORMALLY_ON', () => {
    expect(isLit({
      midiInActivity: false,
      midiOutActivity: false,
      midiActivityLEDMode: ACTIVITY_LED_MODE_NORMALLY_ON,
    })).toBe(true)
    expect(isLit({
      midiInActivity: true,
      midiOutActivity: false,
      midiActivityLEDMode: ACTIVITY_LED_MODE_NORMALLY_ON,
    })).toBe(false)
    expect(isLit({
      midiInActivity: false,
      midiOutActivity: true,
      midiActivityLEDMode: ACTIVITY_LED_MODE_NORMALLY_ON,
    })).toBe(false)
    expect(isLit({
      midiInActivity: true,
      midiOutActivity: true,
      midiActivityLEDMode: ACTIVITY_LED_MODE_NORMALLY_ON,
    })).toBe(false)
  })

  it('lights up correctly for ACTIVITY_LED_MODE_NORMALLY_OFF', () => {
    expect(isLit({
      midiInActivity: false,
      midiOutActivity: false,
      midiActivityLEDMode: ACTIVITY_LED_MODE_NORMALLY_OFF,
    })).toBe(false)
    expect(isLit({
      midiInActivity: true,
      midiOutActivity: false,
      midiActivityLEDMode: ACTIVITY_LED_MODE_NORMALLY_OFF,
    })).toBe(true)
    expect(isLit({
      midiInActivity: false,
      midiOutActivity: true,
      midiActivityLEDMode: ACTIVITY_LED_MODE_NORMALLY_OFF,
    })).toBe(true)
    expect(isLit({
      midiInActivity: true,
      midiOutActivity: true,
      midiActivityLEDMode: ACTIVITY_LED_MODE_NORMALLY_OFF,
    })).toBe(true)
  })

  it('lights up correctly for ACTIVITY_LED_MODE_ALWAYS_ON', () => {
    expect(isLit({
      midiInActivity: false,
      midiOutActivity: false,
      midiActivityLEDMode: ACTIVITY_LED_MODE_ALWAYS_ON,
    })).toBe(true)
    expect(isLit({
      midiInActivity: true,
      midiOutActivity: false,
      midiActivityLEDMode: ACTIVITY_LED_MODE_ALWAYS_ON,
    })).toBe(true)
    expect(isLit({
      midiInActivity: false,
      midiOutActivity: true,
      midiActivityLEDMode: ACTIVITY_LED_MODE_ALWAYS_ON,
    })).toBe(true)
    expect(isLit({
      midiInActivity: true,
      midiOutActivity: true,
      midiActivityLEDMode: ACTIVITY_LED_MODE_ALWAYS_ON,
    })).toBe(true)
  })

  it('lights up correctly for ACTIVITY_LED_MODE_ALWAYS_OFF', () => {
    expect(isLit({
      midiInActivity: false,
      midiOutActivity: false,
      midiActivityLEDMode: ACTIVITY_LED_MODE_ALWAYS_OFF,
    })).toBe(false)
    expect(isLit({
      midiInActivity: true,
      midiOutActivity: false,
      midiActivityLEDMode: ACTIVITY_LED_MODE_ALWAYS_OFF,
    })).toBe(false)
    expect(isLit({
      midiInActivity: false,
      midiOutActivity: true,
      midiActivityLEDMode: ACTIVITY_LED_MODE_ALWAYS_OFF,
    })).toBe(false)
    expect(isLit({
      midiInActivity: true,
      midiOutActivity: true,
      midiActivityLEDMode: ACTIVITY_LED_MODE_ALWAYS_OFF,
    })).toBe(false)
  })

  it('should utilize mapStateToProps', () => {
    const state = { shifter }
    expect(mapStateToProps(state)).toEqual(state)
  })
})
