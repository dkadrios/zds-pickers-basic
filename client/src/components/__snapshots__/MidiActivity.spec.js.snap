// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MidiActivity tests renders correctly 1`] = `
<div
  style={
    Object {
      "margin": "8px 7px 0 0",
    }
  }
>
  <Led
    isMidiActivity={true}
    label="label"
    layout="left"
    lit={true}
  />
</div>
`;
